### DISCUSSION TITLE
The dokku template nginx.conf.sigil contains parameters that are obsolate or deprecated in nginx 1.26.1

### DISCUSSION THREAD

**Discussion by eager-2-learn** [dokku/dokku#7047] [Q&A]
We use dokku 0.34.7 and nginx 1.26.1

Dokku contains the template /var/lib/dokku/plugins/available/nginx-vhosts/templates/nginx.conf.sigil
From this template the nginx.conf file is generated.
The generated nginx.conf contains the parameter: "http2_push_preload".
But the command nginx -t give the warning:
         nginx: [warn] the "http2_push_preload" directive is obsolete, ignored in /home/dokku/nextcloud/nginx.conf:41
         
The generated nginx.conf contains also the line: "listen      443 ssl http2;"
But the command nginx -t give the warning:
        nginx: [warn] the "listen ... http2" directive is deprecated, use the "http2" directive instead in /home/dokku/ov-ws/nginx.conf
 
The new syntax is:
       listen     ssl 443;
       http2 on;
       
Is there a new template available that generates an nginx.conf that does not contain rules that are obsolete or deprecated
for nginx 1.26.1?

**Comment by josegonzalez** [comment]
There is only the one bundled with Dokku. What version of nginx started spitting out those errors, and what is the earliest version that is compatible with the new syntax?

**Comment by eager-2-learn** [comment]
We don't use all different version of nginx.
We only know that Nginx 1.18.0 accept the syntax generated by Dokku and that Nginx 1.26.1 give warning's.
Maybe is it possible to change the template myself?

**Reply by josegonzalez** [reply]
Can you check the nginx changelog to see when the change to deprecate/warn was introduced?

**Comment by eager-2-learn** [comment]
In the Nginx verion 1.25.1 the deprecate/obsolete warning was introduced.
 https://forum.hestiacp.com/t/nginx-1-25-1-listen-http2-directive-is-deprecated/9816
 http://nginx.org/en/docs/http/ngx_http_v2_module.html

**Comment by eager-2-learn** [comment]
Josegonzalez
Is my information sufficient?
Can you give a reaction?

**Reply by josegonzalez** [reply]
> Note: you don't need to make a new comment each time, you can just add a reply to an existing thread here.

We can probably detect if that version is in use, but I would like to say that the current LTS versions of Ubuntu _do not_ use that newer version. What operating system are you running?

**Comment by eager-2-learn** [comment]
We use as operating system Ubuntu 20.04.6
On Ubuntu 20.04.6 give the command "apt policy nginx" "Candidate: 1.18.0-0ubuntu1.5"
But Nginx 1.18.0 has security problems.
Is it possible to change the template myself?
Can you explain how the template is constructed and how I can change it myself?

**Reply by josegonzalez** [reply]
Ubuntu backports security patches to all LTS releases, so that 0ubuntu1.5 probably includes whatever patch you need.

You can customize the nginx.conf.sigil via this method: https://dokku.com/docs/networking/proxies/nginx/#customizing-the-nginx-configuration
